### PR TITLE
CV UI - Promote, Publish, Add Content 

### PR DESCRIPTION
**Pull request by sambible**
This PR is adding support for Publishing CVs, Promoting CVs, Adding Content to CVs, among other changes that are required to bring this work up to current formatting standards, as well as function properly. 

Testing of various methods added here in this PR: https://github.com/SatelliteQE/robottelo/pull/16238

### CODE REVIEW

**Comment by sambible** [comment]
New plan, leaving all the stuff here, and raising a new publish PR

File: airgun/entities/contentview_new.py [hunk 1]
```diff
@@ -22,6 +29,57 @@ def search(self, value):
...
+        view.progressbar.wait_for_result()
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        return view.versions.table.read()
+
+    def update(self, entity_name, values):
+        """Update existing content view"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        # need a wait to recognize the loading is complete
+        # sleep works for now
+        time.sleep(3)
```
Comment (omkarkhatavkar) [inline airgun/entities/contentview_new.py:77]: no hard sleep!

**Comment by omkarkhatavkar** [comment]
@sambible if this PR is going to be included in the Publish PR https://github.com/SatelliteQE/airgun/pull/948 , please close this one

File: airgun/entities/contentview_new.py (hunk 1 above)
Comment (sambible) [reply airgun/entities/contentview_new.py:77]: this was the old me omkar, I can fix it :)

**Comment by sambible** [comment]
This is going to stay in draft for a while, and Publish will get merged. I want to hold my Promote work here, but I jumped the gun on working on it, so I'll come back in a few sprints and revisit this one. :)

**Comment by Gauravtalreja1** [comment]
@sambible bump

**Comment by shweta83** [comment]
@sambible Any update on this one?

**Comment by sambible** [comment]
@shweta83 active development again, should have an update on this soon. Hopefully I can get it ready somewhat quickly, and we can get your e2e tests unblocked.

**Comment by sambible** [comment]
Hoping to get some eyes on this soon, fixing this PR up and adding more support for things. Might be a bear to review, especially considering the age of some of the code in here ( Lai Tran wrote some portions of this ) 

Looking to demonstrate this working with a few end - to end test cases coming very soon.

**Comment by shweta83** [comment]
Can you please provide some more details in the description about the PR. It would be helpful to review the PR.

File: airgun/entities/contentview_new.py [hunk 2]
```diff
@@ -34,20 +40,57 @@ def search(self, value):
...
+            view.always_update.fill(True)
+        if version:
+            view.version_select.item_select(version)
+        view.submit_button.click()
+        view = self.navigate_to(self, 'Edit', entity_name=ccv_name)
+        return view.content_views.resources.read()
+
+    def read_cv(self, entity_name, version_name):
+        """Reads the table for a specified Content View's specified Version"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
```
Comment (damoore044) [inline airgun/entities/contentview_new.py:85]:
```suggestion
        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
        self.browser.plugin.ensure_page_safe(timeout='5s')
        view.wait_displayed()
```
I ran into intermittent local & PRT issue finding `view.versions` for search. 
Waiting until displayed resolved the issue from my draft Robo [PR 16283](https://github.com/SatelliteQE/robottelo/pull/16283),
 ( using draft airgun #1592, a rebase of this pr ) My draft's PRT still failed but not on `read_cv` anymore

File: airgun/entities/contentview_new.py (hunk 1 above)
Comment (sambible) [reply airgun/entities/contentview_new.py:194]: returning to this: have tried various other things, and the sleep is the only thing consistently working

**Review by damoore044 — COMMENTED** [review]
ack, quick comment about `read_cv` method from local use and PRT

File: airgun/entities/contentview_new.py (hunk 1 above)
Comment (LadislavVasina1) [reply airgun/entities/contentview_new.py:196]: I would say keep it then if there is no other way. 
As I believe in many cases there is unfortunately none. :/

Comment (LadislavVasina1) [reply airgun/entities/contentview_new.py:196]: And then I would remove `# sleep works for now` comment :)

**Review by damoore044 — APPROVED** [review]

**Pull request merged**